Mobile Buy SDK/Mobile Buy SDK/Payment Providers/BUYApplePayPaymentProvider.m

```diff
@@ -242,7 +242,7 @@ - (void)paymentAuthorizationViewController:(PKPaymentAuthorizationViewController
 			default:
 				break;
 		}
-		
+		self.checkout = self.applePayAuthorizationDelegate.checkout;
```

Comment:
I think we should just assume that the checkout on the delegate is always the most recent one. Moreover, the checkout on the payment provider should probably just be a cover for that of the delegate. KVO seems unnecessary when the delegate is a private implementation of the provider, and, frankly, needs to be removed in future.